### PR TITLE
WIP: Get total runs

### PR DESCRIPTION
**Pull request by djangoliv**

### CODE REVIEW

File: treeherder/intermittents_commenter/fetch.py [hunk 1]
```diff
@@ -22,3 +22,9 @@ def fetch_testrun_matrix():
     testrun_matrix_url = f"{firefoxci_artefact_api_url}/test-info-testrun-matrix.json"
     response = requests.get(testrun_matrix_url, headers={"User-agent": "mach-test-info/1.0"})
     return response.json()
+
+
+def get_summary_groups():
+    url = "https://treeherder.mozilla.org/api/groupsummary/"
+    response = requests.get(url, headers={"User-agent": "mach-test-info/1.0"})
```
Comment (jmaher) [inline treeherder/intermittents_commenter/fetch.py:29]: I believe this only gets a single day, we will want an entire range of data.  Also this takes ~25 seconds to return, which is why there is an artifact that exists with this data already, and it has 30 days of history, so you can choose which days you want. 

The only problem is you will need to combine data from different days to summarize it.

File: treeherder/intermittents_commenter/commenter.py [hunk 2]
```diff
@@ -385,6 +387,19 @@ def get_bug_run_info(self, bug):
...
+        tasks_and_count = {}
+        summary_groups = (
+            fetch.get_summary_groups() if self.summary_groups is None else self.summary_groups
+        )
+        all_task_labels = summary_groups["job_type_names"]
+        for tasks_by_manifest in summary_groups["manifests"]:
+            for man, tasks in tasks_by_manifest.items():
+                if manifest == man:
+                    for task_index, status, _, count in tasks:
+                        tasks_and_count.setdefault(all_task_labels[task_index], {})[status] = count
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:400]: if we look at >1 day we could end up accumulating data here, so it will need to be a `+= count` and also assigned to `= 0` somewhere else.

in many ways we don't care about the `status`, mostly just number of jobs run :)  it is probably better to split on `status` because that can allow for other calculations, etc.

File: treeherder/intermittents_commenter/fetch.py (hunk 1 above)
Comment (djangoliv) [reply treeherder/intermittents_commenter/fetch.py:29]: OK, thanks for that precisions.
Can you give me the url of the summary group artifact please? and how to access a particular day?

Comment (jmaher) [reply treeherder/intermittents_commenter/fetch.py:29]: test-run-info.json: https://searchfox.org/mozilla-central/source/testing/testinfo.py#400

there is code in there to get the data and squash it to combine days into a larger data structure:
https://searchfox.org/mozilla-central/source/testing/testinfo.py#397

as this is run counts split up by day, when looking at a week (7 days) you will need to squash data, otherwise you can pull a specific date (current day - 1) and get results.

File: treeherder/intermittents_commenter/commenter.py [hunk 3]
```diff
@@ -413,22 +441,30 @@ def build_bug_map(self, bugs, option_collection_map):
...
-                self.testrun_matrix = testrun_matrix
-                bug_testrun_matrix = testrun_matrix[manifest]
+            run_count = 0
+            if test_name:
+                manifest = all_tests[test_name][0]
+                if manifest:
+                    tasks_count = self.get_task_labels_and_count(manifest, start_day, end_day)
+                    job_name = bug["job__signature__job_type_name"]
+                    for task_name, count in tasks_count.items():
+                        if task_name in job_name:
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:458]: this logic is right, but there is an edge case when dealing with variants, lets say you have:
`test-linux/debug-mochitest-1`
`test-linux/debug-mochitest-swr-1`
`test-linux/debug-mochitest-swr-nofis-1`
`test-linux/debug-mochitest-spi-nw-1`

so here if you are starting with a taskname `test-linux/debug-mochitest`, it will match in all the above job_name.  you could do one of 2 things:
1) get list of variants and sort desc, find the best fit match
2) match on length, job_name will need to strip the `-1` at the end (if it exists) and then you could do `if task_name == job_name`

File: treeherder/intermittents_commenter/commenter.py [hunk 4]
```diff
@@ -447,7 +483,10 @@ def build_bug_map(self, bugs, option_collection_map):
                 bug_infos.total = 1
                 bug_infos.test_variants |= all_variants
                 bug_infos.per_repositories[repo] = 1
-                bug_infos.data_table[platform_and_build] = {test_variant: 1}
+                bug_infos.data_table[platform_and_build] = {
+                    test_variant: 1,
+                    "total_runs": run_count,
+                }
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:489]: maybe to support the variants, it could be:
`bug_infos.data_table[platform_and_build] = {'test_variant': {'count': 1, 'runs': run_count}}`

**Review by jmaher — CHANGES_REQUESTED** [review]
this seems to be getting closer.  I imagine once this is approved and landed there might be some small tweaks to handle edge cases.  I am looking to make sure this:
 * works in general, errors out and goes to existing mode when not a general case
 * doesn't use a lot of cpu/memory

File: treeherder/intermittents_commenter/commenter.py [hunk 5]
```diff
@@ -413,22 +441,30 @@ def build_bug_map(self, bugs, option_collection_map):
...
-                bug_testrun_matrix = testrun_matrix[manifest]
+            run_count = 0
+            if test_name:
+                manifest = all_tests[test_name][0]
+                if manifest:
+                    tasks_count = self.get_task_labels_and_count(manifest, start_day, end_day)
+                    job_name = bug["job__signature__job_type_name"]
+                    for task_name, count in tasks_count.items():
+                        if task_name in [job_name, job_name.rsplit("-", 1)[0]]:
+                            run_count = count
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:459]: an edge case here is:
task_name=test-linux1804-64/debug-mochitest-plain-swr
job_name=test-linux1804-64/debug-mochitest-plain-swr-nofis

so task_name in job_name -> True;  this will end up being misleading.  Maybe to solve this, see if `task_name==job_name or task_name==job_name.rsplit(...)`

File: treeherder/intermittents_commenter/commenter.py [hunk 6]
```diff
@@ -459,9 +497,10 @@ def build_bug_map(self, bugs, option_collection_map):
                 data_table = bug_infos.data_table
                 platform_and_build_data = data_table.get(platform_and_build, {})
                 data_table[platform_and_build] = platform_and_build_data
-                data_table[platform_and_build][test_variant] = (
-                    platform_and_build_data.get(test_variant, 0) + 1
-                )
+                if test_variant in data_table[platform_and_build]:
+                    data_table[platform_and_build][test_variant]["count"] += 1
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:501]: what scenario would the count increase?  is this safe to know we are counting all of the run_count properly?

**Review by jmaher — CHANGES_REQUESTED** [review]
this is getting real close.  I know there is a lot of specific code, we can work to make the code more modular with other tools when the functionality is working as expected.

File: treeherder/intermittents_commenter/commenter.py [hunk 7]
```diff
@@ -498,6 +498,7 @@ def build_bug_map(self, bugs, option_collection_map, start_day, end_day):
                 platform_and_build_data = data_table.get(platform_and_build, {})
                 data_table[platform_and_build] = platform_and_build_data
                 if test_variant in data_table[platform_and_build]:
+                    breakpoint()
```
Comment (jmaher) [inline treeherder/intermittents_commenter/commenter.py:501]: is this leftover from debugging?

Comment (djangoliv) [reply treeherder/intermittents_commenter/commenter.py]: Sorry, I I deleted it
